xdsl/ir.py

```diff
@@ -418,13 +418,12 @@ def __post_init__(self):
         assert (isinstance(self.name, str))
 
     @staticmethod
-    def with_result_types(
-            op: Any,
-            operands: Sequence[SSAValue] | None = None,
-            result_types: Sequence[Attribute] | None = None,
-            attributes: dict[str, Attribute] | None = None,
-            successors: Sequence[Block] | None = None,
-            regions: Sequence[Region] | None = None) -> Operation:
+    def with_result_types(op: type[OpT],
```

Comment:
IMO, this should be removed. Not sure whether we do this in this PR or in a later one.

Reply:
Yeah, let's remove this in a later PR, once we understand what we really want.